File: problems/HackerRank/2521_exercises/tree/isSymmetric.c

bool dfs(struct TreeNode *p, struct TreeNode *q)
{
	if (p == NULL && q == NULL)
	{
		return true;
	}

	if ((p == NULL || q == NULL) || (p->val != q->val))
	{
		return false;
	}

	return dfs(p->left, q->right) && dfs(p->right, q->left);
}

bool isSymmetric(struct TreeNode *root)
{
	return dfs(root->left, root->right);
}

File: problems/LeetCode/tree/isSymmetric.py
def isSymmetric(self, root: Optional[TreeNode]) -> bool:

		def isSymmeTrees(left: Optional[TreeNode], right: Optional[TreeNode]) -> bool:

				if not left and not right:
						return True
				
				if (not left or not right) or (left.val != right.val):
						return False
				
				return isSymmeTrees(left.left, right.right) and isSymmeTrees(left.right, right.left)

		
		return isSymmeTrees(root.left, root.right)